### PR TITLE
Use more modern PyGranta-style CI

### LINKED ISSUE #487: Use PyAnsys actions in CI
We can use the following Ansys CI actions:

- Doc build (with and without examples)
- Build wheelhouse
- Build library
- Release
- Deploy stable docs

We should probably just replicate the CI configuration from the JobQueue repository

### PR DESCRIPTION
**Pull request by Andy-Grigg**
Closes #487 

Use more PyAnsys actions in CI. These changes have largely been copied from the grantami-jobqueue repository. These changes required some changes to the tox config and sphinx configuration as well.

Branch protection rules will need to be modified to reflect the change in job/step names before merging.

### CODE REVIEW

**Comment by Andy-Grigg** [comment]
I've checked the artifacts and they all look good. The docs look to have built correctly, including the examples.

File: doc/source/conf.py [hunk 1]
```diff
@@ -15,18 +15,38 @@
...
-author = "ANSYS Inc."
-html_title = f"PyGranta BoM Analytics {__version__}"
+cname = os.getenv("DOCUMENTATION_CNAME", "bomanalytics.grantami.docs.pyansys.com")
+"""The canonical name of the webpage hosting the documentation."""
 
-# The short X.Y version
-release = version = __version__
+# specify the location of your github repo
+html_theme_options = {
+    "github_url": "https://github.com/ansys/grantami-jobqueue",
```
Comment (da1910) [inline doc/source/conf.py:36]:
```suggestion
    "github_url": "https://github.com/ansys/grantami-bomanalytics",
```

**Review by da1910 — CHANGES_REQUESTED** [review]
One link, looks good otherwise

**Review by da1910 — APPROVED** [review]

File: .github/workflows/ci_cd.yml [hunk 2]
```diff
@@ -1,104 +1,168 @@
...
     tags:
-      - '*'
+      - "*"
     branches:
       - main
-    paths-ignore:
-      - '.github/**'
-      - 'pyproject.toml'
   workflow_dispatch:
-  merge_group:
```
Comment (da1910) [inline .github/workflows/ci_cd.yml:15]: We need this merge_group: trigger

Comment (Andy-Grigg) [reply .github/workflows/ci_cd.yml:15]: Thanks!

**Pull request merged**